content/english/blog/dotnet_are_we_webassembly_yet.md

```diff
@@ -0,0 +1,115 @@
+
+ℹ️ **BUT**, one more thing before moving on. It is important to note that the way this stuff works is that the entire .NET Framework core and its runtime is embedded into the `.wasm` module (you can actually see the `dll`s crammed into a `data` section if you open the file up). So while we can expect things to get smaller and more optimized over time, I don't think we'll ever see .NET `.wasm` files approach the smaller relative size of Rust and Zig binaries because those languages don't require a garbage collector, etc.
+
+In the interest of exploration, I wonder what happens when we try and do `dotnet run` (I'm assuming it should run like any other console/stdio-based WASI module).
+
+```
+dotnet run
+Unhandled exception: System.ComponentModel.Win32Exception (2): An error occurred trying to start process 'wasmtime' with working directory '/Users/kevin/lab/dotnet/MyFirstWasiApp'. No such file or directory
+```
+
+Well, that's unfortunate. Looks like the run command is tightly coupled to having the `wasmtime` binary in your path. This is actually a good time for me to point out another area of high friction that I've found with .NET (though the problem exists in TinyGo as well): _lack of runtime interoperability_.
+
+During a day of playing around with building all kinds of different freestanding WASI modules (yes, you can actually do this with the .NET WASI SDK!), I ran into a pretty thorny problem. At times, I would be able to get something to work when using a `wasmtime` host, but it would not work when using `wasmer` or `wasm3` hosts. Conversely, sometimes things would break via `wasmtime` that didn't break via `wasmer`.
+
+_But what about **wasmCloud**?_ When will be able to build .NET actors in wasmCloud? Technically, it's feasible today, but with a giant asterisk ⚠️. It required me creating an elaborate Rube Goldberg machine made up of duct tape, straws, bailing wire, and a couple pieces of chewing gum. I had to hand-craft some free-range organic (antibiotic free!) C header files because you can't control import/export labels from inside a C# file (yet).
+
+There is also absolutely no sign of work beginning in .NET on support for the component model (not to be confused with the `System.ComponentModel` namespace, which has been around since the dinosaurs). Even when the .NET wasm developer experience gets to a point that feels low-friction, it will probably not support components yet. We're definitely going to have to wait (and put in some effort and contribute!) to see `wit`-style support in a .NET project.
```

Comment:
After `There is also absolutely no sign of work beginning in .NET on support for the component model`, this is a great time to link back to the ubiquity blog post and tie this post with a nice pretty bow of "We know the component model isn't there yet, and things are very runtime specific. This is the way it works today, and the team maintaining wasmCloud encourages this standards work so that WebAssembly developers don't have difficulty with different language toolchains"